src/List.svelte

```diff
@@ -0,0 +1,48 @@
+<script lang="ts">
+  import Loading from "@app/Loading.svelte";
+
+  export let items: any[];
+  export let query: any;
+
+  // Being passed to more slot to handle loading state of trigger
+  let loading = false;
+  // Should be changed to true, once no more items are being returned.
+  let complete = false;
+
+  const fetchMore = async () => {
+    loading = true;
+    const response = await query();
+
+    if (response.length > 0) {
+      items.push(...response);
+      // We explicitly do an assignment here to force Svelte to update the slots props
+      items = items;
```

Comment:
Would it be better to split the starting items from the loaded ones?

Reply:
It would be possible, but in some cases like commit history, we ultimately need one single array of items to group them by date.

Reply:
ok, then perhaps `items = [...items, ...response]`?

Reply:
:eyes:

Reply:
Hey thanks for the push, missed this comment, was a tricky one, it seemed that doing a `Array.push()` and then assigning was doing the trick while doing the spread one here not..
In the end I went with binding `items` passed into `List` with the initial list, I think it's a good solution, due to the availability of the data.
The only caveat that I can think of is that the `List` component now is somehow a fetcher and a display component, which initially I wanted to dumb down.